cpp/test/phonenumbers/phonenumbermatcher_test.cc

```diff
@@ -1078,6 +1078,25 @@ TEST_F(PhoneNumberMatcherTest, NoMatchIfNoNumber) {
   EXPECT_FALSE(matcher->HasNext());
 }
 
+TEST_F(PhoneNumberMatcherTest, NoErrorWithSpecialCharacters) {
+  string stringWithSpecialCharacters =
+      "Myfuzzvar1152: \"My info:%415-666-7777 123 fake street\"\nfuzzvar1155: "
+      "47\nfuzzvar1158: %415-666-1234 "
+      "i18n_phonenumbers_Pho\356eNumberMatcher_Leniency_VALID_1"
+      "\nfuzzvar1159: 20316 info:%415-666-7777 123 fake str79ee\nt";
+  string Numbers;
+  for (int i = 0; i < 100; ++i)
+    Numbers.append(stringWithSpecialCharacters);
+  scoped_ptr<PhoneNumberMatcher> matcher(
+      GetMatcherWithLeniency(Numbers, RegionCode::US(),
+                             PhoneNumberMatcher::POSSIBLE));
+  PhoneNumberMatch match;
+  while (matcher->HasNext()) {
+    matcher->Next(&match);
+  }
+  EXPECT_TRUE(true);
```

Comment:
EXPECT_TRUE(true)? :)

Reply:
My bad. UT is not updated, checked only through debug results.
Updated now.